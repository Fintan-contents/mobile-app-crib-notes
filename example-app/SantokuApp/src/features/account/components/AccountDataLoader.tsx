import {handleErrorWithAlert} from 'bases/core/errors/handleErrorWithAlert';
import {resolveErrorMessage} from 'bases/message/resolveErrorMessage';
import React, {useEffect, useRef} from 'react';

import {useIsLoggedIn} from '../client-states/useIsLoggedIn';
import {useAccountData} from '../services/account/useAccountData';

export const AccountDataLoader: React.FC<React.PropsWithChildren> = ({children}) => {
  const [isLoggedIn] = useIsLoggedIn();
  const isAutoLoggedIn = useRef(isLoggedIn).current;
  const {isLoading, isLoadingError, errorUpdateCount, error} = useAccountData({
    enabled: isLoggedIn,
    meta: {disableGlobalErrorHandler: true},
  });

  useEffect(() => {
    // オートログイン済で初回のデータ取得時にエラーが発生した場合は
    if (isAutoLoggedIn && isLoadingError) {
      if (errorUpdateCount === 1) {
        const {title, message} = resolveErrorMessage(error);
        handleErrorWithAlert(error, title, message);
      }
    }
  }, [error, errorUpdateCount, isAutoLoggedIn, isLoadingError]);

  if (isAutoLoggedIn && (isLoading || isLoadingError)) {
    return null;
  }
  return <>{children}</>;
};
