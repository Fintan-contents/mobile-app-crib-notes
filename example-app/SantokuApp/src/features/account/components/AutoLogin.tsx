import {handleErrorWithAlert} from 'bases/core/errors/handleErrorWithAlert';
import {resolveErrorMessage} from 'bases/message/resolveErrorMessage';
import React, {useEffect} from 'react';

import {useIsLoggedIn} from '../client-states/useIsLoggedIn';
import {isUnauthorizedError} from '../errors/UnauthorizedError';
import {useAuthCommands} from '../services/auth/useAuthCommands';

export const AutoLogin: React.FC<React.PropsWithChildren> = ({children}) => {
  const [isLoggedIn] = useIsLoggedIn();
  const {autoLogin} = useAuthCommands();
  useEffect(() => {
    autoLogin().catch(e => {
      if (!isUnauthorizedError(e)) {
        const {title, message} = resolveErrorMessage(e);
        handleErrorWithAlert(e, title, message);
      }
    });
  }, [autoLogin]);
  // オートログイン時にエラーが発生した場合は、isLoggedInには何も値が設定されない
  // isLoggedInに値が設定されたタイミングでオートログインの処理完了とみなす
  if (isLoggedIn == null) {
    return null;
  }
  return <>{children}</>;
};
