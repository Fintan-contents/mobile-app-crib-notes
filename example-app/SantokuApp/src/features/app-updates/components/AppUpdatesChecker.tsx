import * as Application from 'expo-application';
import React, {useEffect} from 'react';

import {useAppUpdates} from '../services/useAppUpdates';
import {showUpdateRequiredDialog} from '../use-cases/showUpdateRequiredDialog';

const AppUpdatesChecker: React.FC<React.PropsWithChildren> = ({children}) => {
  const {appUpdates} = useAppUpdates();

  useEffect(() => {
    if (appUpdates?.updateRequired) {
      showUpdateRequiredDialog(Application.nativeApplicationVersion!);
    }
  }, [appUpdates]);
  if (appUpdates === undefined || appUpdates.updateRequired) {
    return null;
  }

  return <>{children}</>;
};

export {AppUpdatesChecker};
