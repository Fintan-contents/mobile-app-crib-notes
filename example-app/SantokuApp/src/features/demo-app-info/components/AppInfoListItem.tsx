import React from 'react';
import {Text, View} from 'react-native';
import {ListItem} from 'react-native-elements';

export type AppInfoItemData = {
  name?: string;
  value: string | null;
};

type AppInfoListItemProps = {
  item: AppInfoItemData;
};

export const AppInfoListItem: React.FC<AppInfoListItemProps> = ({item}) => {
  return !item.name ? null : (
    <ListItem bottomDivider>
      <ListItem.Content>
        <ListItem.Title>
          <Text>{item.name}</Text>
        </ListItem.Title>
        <View style={{marginTop: 10, marginLeft: 10}}>
          <Text>{item.value}</Text>
        </View>
      </ListItem.Content>
    </ListItem>
  );
};
