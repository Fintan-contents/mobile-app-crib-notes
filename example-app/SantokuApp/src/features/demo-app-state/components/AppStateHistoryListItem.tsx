import React from 'react';
import {StyleSheet, View} from 'react-native';
import {ListItem, Text} from 'react-native-elements';

import {StateChangedEventRecord} from '../types/StateChangedEventRecord';

type AppStateHistoryListItemProps = {
  item: StateChangedEventRecord;
};

export const AppStateHistoryListItem: React.FC<AppStateHistoryListItemProps> = ({item}) => {
  return (
    <ListItem bottomDivider>
      <View style={styles.record}>
        <View style={styles.stateAndEvent}>
          <Text>{item.event}</Text>
          <Text style={styles.state}>{item.state}</Text>
        </View>
        <Text style={styles.timestamp}>{item.timestamp?.toISOString()}</Text>
      </View>
    </ListItem>
  );
};
const styles = StyleSheet.create({
  record: {flex: 1, flexDirection: 'row', justifyContent: 'space-between'},
  stateAndEvent: {flex: 1, flexDirection: 'row'},
  state: {fontWeight: 'bold', marginLeft: 5},
  timestamp: {color: '#666666'},
});
