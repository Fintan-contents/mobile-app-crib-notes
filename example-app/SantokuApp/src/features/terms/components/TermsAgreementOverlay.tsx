import {FullWindowOverlay} from 'bases/ui/overlay/FullWindowOverlay';
import React, {useCallback, useMemo, useState} from 'react';

import {
  TermsAgreementOverlayComponent,
  TermsAgreementOverlayProps,
  TermsAgreementOverlayShowProps,
} from './TermsAgreementOverlayComponent';

type TermsAgreementOverlayType = {
  show: (props: TermsAgreementOverlayShowProps) => void;
  close: () => void;
  Component: typeof Component;
};

/**
 * 利用規約をOverlay表示するコンポーネント。
 */
const Component: React.FC = () => {
  const [state, setState] = useState<
    Omit<TermsAgreementOverlayProps, 'termsOfService'> & Partial<Pick<TermsAgreementOverlayProps, 'termsOfService'>>
  >({
    visible: false,
    close: () => {},
  });
  const close = useCallback(() => setState(prevState => ({...prevState, visible: false})), []);
  const show = useCallback(
    (props: Omit<TermsAgreementOverlayProps, 'visible' | 'close'>) => {
      setState({
        ...props,
        visible: true,
        close,
      });
    },
    [close],
  );

  TermsAgreementOverlay = useMemo(
    () => ({
      ...TermsAgreementOverlay,
      show,
      close,
    }),
    [close, show],
  );

  return (
    <FullWindowOverlay>
      {state.termsOfService && <TermsAgreementOverlayComponent {...state} termsOfService={state.termsOfService} />}
    </FullWindowOverlay>
  );
};

let TermsAgreementOverlay: TermsAgreementOverlayType = {
  show: (props: TermsAgreementOverlayShowProps) => {
    throw new Error('TermsAgreementOverlay.Component is required.');
  },
  close: () => {
    throw new Error('TermsAgreementOverlay.Component is required.');
  },
  Component,
};

export {TermsAgreementOverlay};
