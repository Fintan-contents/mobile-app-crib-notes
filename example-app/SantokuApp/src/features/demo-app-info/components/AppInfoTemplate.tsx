import React from 'react';
import {FlatList} from 'react-native';

import {AppInfoItemData, AppInfoListItem} from './AppInfoListItem';

type AppInfoTemplateProps = {
  items: AppInfoItemData[];
  keyExtractor: (item: AppInfoItemData, index: number) => string;
  testID?: string;
};

export const AppInfoTemplate: React.FC<AppInfoTemplateProps> = ({items, keyExtractor, testID}) => {
  return <FlatList data={items} renderItem={AppInfoListItem} keyExtractor={keyExtractor} testID={testID} />;
};
