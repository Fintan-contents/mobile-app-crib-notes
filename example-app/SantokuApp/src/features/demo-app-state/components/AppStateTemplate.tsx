import React, {useRef} from 'react';
import {FlatList} from 'react-native';

import {AppStateHistoryListItem} from './AppStateHistoryListItem';
import {StateChangedEventRecord} from '../types/StateChangedEventRecord';

type AppStateTemplateProps = {
  history: StateChangedEventRecord[];
  keyExtractor: (item: StateChangedEventRecord, index: number) => string;
  testID?: string;
};

export const AppStateTemplate: React.FC<AppStateTemplateProps> = ({history, keyExtractor, testID}) => {
  const listRef = useRef<FlatList<StateChangedEventRecord>>(null);
  return (
    <FlatList
      ref={listRef}
      data={history}
      renderItem={({item}) => <AppStateHistoryListItem item={item} />}
      keyExtractor={keyExtractor}
      onContentSizeChange={() => listRef.current?.scrollToEnd()}
      testID={testID}
    />
  );
};
