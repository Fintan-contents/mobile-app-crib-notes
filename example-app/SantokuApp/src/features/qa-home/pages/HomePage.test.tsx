import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import {BundledMessagesLoader} from 'bases/message/BundledMessageLoader';
import {loadMessages} from 'bases/message/Message';
import {Snackbar} from 'bases/ui/snackbar/Snackbar';
import {AppThemeProvider} from 'bases/ui/theme/AppThemeProvider';
import {TermsAgreementOverlay} from 'features/terms/components/TermsAgreementOverlay';
import {initialData} from 'fixtures/msw/datas';
import {db, initialDb} from 'fixtures/msw/db';
import {handlers} from 'fixtures/msw/handlers';
import {setupServer} from 'msw/node';
import React from 'react';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {QueryClient, QueryClientProvider} from 'react-query';

import type {HomePageProps} from './HomePage';

const server = setupServer(...handlers);
beforeAll(async () => {
  await loadMessages(new BundledMessagesLoader());
  initialDb();
  await initialData();
  db.loggedInAccount.create({accountId: 'santoku'});
  server.listen();
});
afterAll(() => server.close());

beforeEach(() => {
  jest.useFakeTimers();
});
jest.mock('bases/date/formatDiffInDaysOrHours', () => {
  return {
    formatDiffInDaysOrHours: jest.fn(() => {
      return '1時間前';
    }),
  };
});

const Wrapper: React.FC<React.PropsWithChildren> = ({children}) => {
  const queryClient = new QueryClient();
  return (
    <SafeAreaProvider>
      <AppThemeProvider>
        <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
        <TermsAgreementOverlay.Component />
        <Snackbar.Component />
      </AppThemeProvider>
    </SafeAreaProvider>
  );
};

describe('Home', () => {
  it('マウントされたときに正常にレンダリングされること', () => {
    // メッセージのロード前にHomePageをimportしてしまうとエラーになるため、メッセージのロード後にrequireします
    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
    const HomePage = require('./HomePage').HomePage as React.FC<React.PropsWithChildren<HomePageProps>>;
    render(
      <HomePage navigateToQuestionDetail={() => {}} navigateToQuestionAndEventPost={() => {}} setHeader={() => {}} />,
      {
        wrapper: Wrapper,
      },
    );
    expect(screen.findByTestId('HomePage')).not.toBeNull();
    expect(screen).toMatchSnapshot();
  });
});
