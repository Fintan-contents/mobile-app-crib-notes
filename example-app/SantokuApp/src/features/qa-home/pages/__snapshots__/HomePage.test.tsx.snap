// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home マウントされたときに正常にレンダリングされること 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
    testID="HomePage"
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={
        <React.Fragment>
          <ForwardRef
            px="p24"
            py="p32"
          >
            <ForwardRef
              letterSpacing={0.18}
              lineHeight={24}
              variant="font20Bold"
            >
              募集中のイベント
            </ForwardRef>
          </ForwardRef>
          <StyledRow
            alignItems="center"
            justifyContent="space-between"
            px="p24"
            py="p32"
          >
            <ForwardRef
              letterSpacing={0.18}
              lineHeight={24}
              variant="font20Bold"
            >
              質問
            </ForwardRef>
            <StyledRow
              alignItems="center"
              space="p32"
            >
              <ForwardRef
                onPress={[Function]}
              >
                <SortIllustration
                  color="black"
                />
              </ForwardRef>
              <ForwardRef
                onPress={[Function]}
              >
                <FilterAltIllustration />
              </ForwardRef>
              <ForwardRef
                onPress={[Function]}
              >
                <LocalOfferIllustration
                  color="black"
                />
              </ForwardRef>
            </StyledRow>
          </StyledRow>
        </React.Fragment>
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {},
        ]
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <RCTRefreshControl />
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "paddingHorizontal": 24,
                  "paddingVertical": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#212121",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 0.18,
                    "lineHeight": 24,
                  },
                ]
              }
            >
              募集中のイベント
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 24,
                  "paddingVertical": 32,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#212121",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 0.18,
                    "lineHeight": 24,
                  },
                ]
              }
            >
              質問
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "height": 24,
                          "width": 24,
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M3 18H9V16H3V18ZM3 6V8H21V6H3ZM3 13H15V11H3V13Z"
                      fill={4280361249}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingLeft": 32,
                    },
                  ]
                }
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "height": 24,
                          "width": 24,
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M6.99999 6H17L11.99 12.3L6.99999 6ZM4.24999 5.61C6.26999 8.2 9.99999 13 9.99999 13V19C9.99999 19.55 10.45 20 11 20H13C13.55 20 14 19.55 14 19V13C14 13 17.72 8.2 19.74 5.61C20.25 4.95 19.78 4 18.95 4H5.03999C4.20999 4 3.73999 4.95 4.24999 5.61Z"
                      fill={4280361249}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "paddingLeft": 32,
                    },
                  ]
                }
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight={24}
                  bbWidth={24}
                  fill="none"
                  focusable={false}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      Array [
                        Object {
                          "height": 24,
                          "width": 24,
                        },
                      ],
                      Object {
                        "flex": 0,
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                  vbHeight={24}
                  vbWidth={24}
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="M21.41 11.58L12.41 2.58C12.05 2.22 11.55 2 11 2H4C2.9 2 2 2.9 2 4V11C2 11.55 2.22 12.05 2.59 12.42L11.59 21.42C11.95 21.78 12.45 22 13 22C13.55 22 14.05 21.78 14.41 21.41L21.41 14.41C21.78 14.05 22 13.55 22 13C22 12.45 21.77 11.94 21.41 11.58ZM13 20.01L4 11V4H11V3.99L20 12.99L13 20.01Z"
                      fill={4280361249}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                    <RNSVGPath
                      d="M6.5 8C7.32843 8 8 7.32843 8 6.5C8 5.67157 7.32843 5 6.5 5C5.67157 5 5 5.67157 5 6.5C5 7.32843 5.67157 8 6.5 8Z"
                      fill={4280361249}
                      propList={
                        Array [
                          "fill",
                        ]
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": 32,
            "flexDirection": "column",
            "justifyContent": "center",
            "position": "absolute",
            "right": 8,
          },
        ]
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#29B6F6",
                "borderRadius": 28,
                "elevation": 1,
                "flexDirection": "row",
                "height": 56,
                "justifyContent": "center",
                "margin": 8,
                "shadowColor": "#212121",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.35,
                "shadowRadius": 4,
                "width": 56,
              },
            ]
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            fill="none"
            focusable={false}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Array [
                  Object {
                    "height": 24,
                    "width": 24,
                  },
                ],
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGPath
                  d="M19 13H13V19H11V13H5V11H11V5H13V11H19V13Z"
                  fill={4294967295}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                />
              </RNSVGGroup>
              <RNSVGDefs>
                <RNSVGClipPath
                  name="clip0_5404_1261"
                >
                  <RNSVGRect
                    fill={4294967295}
                    height="24"
                    propList={
                      Array [
                        "fill",
                      ]
                    }
                    width="24"
                    x={0}
                    y={0}
                  />
                </RNSVGClipPath>
              </RNSVGDefs>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "position": "absolute",
            "right": 10,
            "top": 10,
          },
        ]
      }
    >
      <ActivityIndicator
        color="#E0E0E0"
        style={
          Array [
            Object {},
          ]
        }
      />
    </View>
  </View>,
  <View
    testID="FullWindowOverlay"
  />,
  <View
    testID="FullWindowOverlay"
  />,
]
`;
